packages/client/src/Components/Mapping/helpers/MapPlanningPlayerListener.js

```diff
@@ -58,30 +43,105 @@ export default class MapPlanningPlayerListener {
       opacity: 0.2
     }
 
-    // try to create a button
+    // If we're using this module in a test class, we can drop out at this point if we don't have
+    // an actual map object
+
+    // create our two lines, one for planning, one for history
+    this.routeLine = L.polyline([], {
+      color: '#fff',
+      dashArray: [1, 4]
+    })
+    this.plannedLine = L.polyline([], {
+      color: '#0f0'
+    })
+
+    // put them on the map
+    this.routeLine.addTo(map)
+    this.plannedLine.addTo(map)
+
+    // use layer groups to store data - so we can confidently remove them
+    this.waypointMarkers = L.layerGroup().addTo(map)
+
+    // TODO: drop these fake state change triggers
     const context = this
-    this.createEasyButtonFor(map, 'fa-dice-one', 'immobile state selected', function (btn, map) {
+    this.btn1aImmobile = createButton(false, '1a. immobile state', function (btn, map) {
```

Comment:
It would be preferable to do the refactor replace `context` with `this`. 

I have refactored this in another PR though so it can be ignored for now.